server/src/main/kotlin/org/javacs/kt/KotlinTextDocumentService.kt

```diff
@@ -254,27 +249,22 @@ class KotlinTextDocumentService(
 
     private fun lintLater(uri: URI) {
         lintTodo.add(uri)
-        if (!linting) {
-            debounceLint.submit(::doLint)
-        }
+        debounceLint.submit(::doLint)
     }
 
     private fun lintNow(file: URI) {
         lintTodo.add(file)
         debounceLint.submitImmediately(::doLint)
     }
 
-    private fun doLint() {
-        linting = true
-        try {
-            LOG.info("Linting {}", describeURIs(lintTodo))
-            val files = clearLint()
-            val context = sp.compileFiles(files)
+    private fun doLint(cancelCallback : () -> Boolean) {
+        LOG.info("Linting {}", describeURIs(lintTodo))
+        val files = clearLint()
+        val context = sp.compileFiles(files)
+        if (!cancelCallback.invoke()) {
```

Comment:
This (the cancellation) is trying to avoid a case where the user pauses typing, sp.compileFiles is called, but before it ends, the user resumes typing - there is, I think, not much point to report the diagnostics in such a case, as they are for an older version of the document's content. The lintting should run again on the new version anyway.

Reply:
Looks good, this should help prevent outdated diagnostics.